### PR TITLE
Update Set-AzureRmSqlServerAuditing and Set-AzureRmSqlDatabaseAuditing Cmdlets

### PR DESCRIPTION
**Pull request by bashahee**
## Description

Adding a new parameter (-StorageAccountSubscriptionId) existing Set-AzureRmSqlServerAuditing and Set-AzureRmSqlDatabaseAuditing Cmdlets.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by praries880** [comment]
@bashahee 
* netcore build has [failing tests](https://travis-ci.org/Azure/azure-powershell/builds/391855690#L1787)
* you need to add a change log that described the changes in the PR.

**Comment by praries880** [comment]
Design review : https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/99

File: src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/AuditingSettings/SetAzureSqlServerAuditing.cs [hunk 1]
```diff
@@ -54,6 +54,13 @@ public class SetAzureSqlServerAuditing : SqlServerAuditingSettingsCmdletBase
         [ValidateNotNullOrEmpty]
         public string StorageAccountName { get; set; }
 
+        /// <summary>
+        /// Gets or sets storage account subscription id.
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = AuditingHelpMessages.AuditStorageAccountSubscriptionIdHelpMessage)]
```
Comment (praries880) [inline src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/AuditingSettings/SetAzureSqlServerAuditing.cs]: As per the discussion in the [design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/99#issuecomment-397068036)

The behavior as the changes stand currently is that the user can specify `StorageAccountSubscriptionId` without specifying `StorageAccountName`.

Given that `StorageAccountName` is required when StorageAccountSubscriptionId` is specified, I would recomment adding a parameter set where `StorageAccountName` is required and `StorageAccountSubscriptionId` is optional.

File: src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/AuditingSettings/SetAzureSqlDatabaseAuditing.cs [hunk 2]
```diff
@@ -60,6 +60,13 @@ public class SetAzureSqlDatabaseAuditing : SqlDatabaseAuditingSettingsCmdletBase
         [ValidateNotNullOrEmpty]
         public string StorageAccountName { get; set; }
 
+        /// <summary>
+        /// Gets or sets storage account subscription id.
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = AuditingHelpMessages.AuditStorageAccountSubscriptionIdHelpMessage)]
```
Comment (praries880) [inline src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/AuditingSettings/SetAzureSqlDatabaseAuditing.cs]: As per the discussion in the [design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/99#issuecomment-397068036)

The behavior as the changes stand currently is that the user can specify `StorageAccountSubscriptionId` without specifying `StorageAccountName`.

Given that `StorageAccountName` is required when StorageAccountSubscriptionId` is specified, I would recomment adding a parameter set where `StorageAccountName` is required and `StorageAccountSubscriptionId` is optional.

**Comment by bashahee** [comment]
@praries880 Your requests were addressed.

File: src/ResourceManager/Sql/Commands.Sql/Common/AzureEndpointsCommunicator.cs [hunk 3]
```diff
@@ -231,5 +232,140 @@ private ResourceManagementClient GetCurrentResourcesClient(IAzureContext context
...
+        private async Task<Dictionary<StorageKeyKind, string>> RetrieveStorageKeysAsync(string storageAccountId)
+        {
+            bool isClassicStorage = storageAccountId.Contains("Microsoft.ClassicStorage/storageAccounts");
+
+            // Build a URI for calling corresponding REST-API
+            //
+            StringBuilder uriBuilder = new StringBuilder(Context.Environment.GetEndpointAsUri(AzureEnvironment.Endpoint.ResourceManager).ToString());
+            uriBuilder.AppendFormat("{0}/listKeys?api-version={1}",
+                storageAccountId,
+                isClassicStorage ? "2016-11-01" : "2017-06-01");
```
Comment (praries880) [inline src/ResourceManager/Sql/Commands.Sql/Common/AzureEndpointsCommunicator.cs]: Nitpick : These string literals should be constants instead.

File: src/ResourceManager/Sql/Commands.Sql/Common/AzureEndpointsCommunicator.cs [hunk 4]
```diff
@@ -231,5 +232,140 @@ private ResourceManagementClient GetCurrentResourcesClient(IAzureContext context
...
+
+            // Extract keys out of response.
+            //
+            Dictionary<StorageKeyKind, string> storageAccountKeys = new Dictionary<StorageKeyKind, string>();
+            string primaryKey = null;
+            string secondaryKey = null;
+            if (isClassicStorage)
+            {
+                primaryKey = (string)storageAccountKeysResponse["primaryKey"];
+                secondaryKey = (string)storageAccountKeysResponse["secondaryKey"];
```
Comment (praries880) [inline src/ResourceManager/Sql/Commands.Sql/Common/AzureEndpointsCommunicator.cs]: Nitpick : These string literals should be constants instead.

File: src/ResourceManager/Sql/Commands.Sql/Common/AzureEndpointsCommunicator.cs (hunk 3 above)
Comment (bashahee) [reply src/ResourceManager/Sql/Commands.Sql/Common/AzureEndpointsCommunicator.cs]: @praries880 Versions were extracted to constants, while Uri template was not due to readability.

File: src/ResourceManager/Sql/Commands.Sql/Common/AzureEndpointsCommunicator.cs (hunk 4 above)
Comment (bashahee) [reply src/ResourceManager/Sql/Commands.Sql/Common/AzureEndpointsCommunicator.cs]: @praries880 Done.

**Comment by bashahee** [comment]
@praries880 Since all requests were addressed and relevant open issue was accepted, can "cmdlet review required" and "needs revision" Labels be removed from PR?

**Comment by praries880** [comment]
on demand : https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/943/

**Pull request merged**